### ISSUE TITLE
Cannot build with GoogleAds plugin (Multiple dex files define)

### ISSUE THREAD

**Issue by ocveran** [playgameservices/play-games-plugin-for-unity#2256]
Hi there!
I created empty poject then imported gpg plugin and google ads plugin. But I cannot build. I get this conflict:

```
CommandInvokationFailure: Gradle build failed. 
C:/Program Files/Java/jdk1.8.0_131\bin\java.exe -classpath "C:\Program Files\Unity\Hub\Editor\2018.2.1f1\Editor\Data\PlaybackEngines\AndroidPlayer\Tools\gradle\lib\gradle-launcher-4.2.1.jar" org.gradle.launcher.GradleMain "-Dorg.gradle.jvmargs=-Xmx2048m" "assembleRelease"

stderr[

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':transformDexWithDexForRelease'.
> com.android.build.api.transform.TransformException: com.android.ide.common.process.ProcessException: java.util.concurrent.ExecutionException: com.android.dex.DexException: Multiple dex files define Lcom/google/android/gms/common/api/internal/zzbr;

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

* Get more help at https://help.gradle.org

BUILD FAILED in 3s
]
stdout[
:preBuild UP-TO-DATE
:GoogleMobileAdsPlugin:preBuild UP-TO-DATE
:GoogleMobileAdsPlugin:preReleaseBuild UP-TO-DATE
:GoogleMobileAdsPlugin:checkReleaseManifest
:GoogleMobileAdsPlugin:processReleaseManifest
:GooglePlayGamesManifest.plugin:preBuild UP-TO-DATE
:GooglePlayGamesManifest.plugin:preReleaseBuild UP-TO-DATE
:GooglePlayGamesManifest.plugin:checkReleaseManifest
:GooglePlayGamesManifest.plugin:processReleaseManifest
:preReleaseBuild
:GoogleMobileAdsPlugin:compileReleaseAidl
:GooglePlayGamesManifest.plugin:compileReleaseAidl
:compileReleaseAidl UP-TO-DATE
:GoogleMobileAdsPlugin:packageReleaseRenderscript NO-SOURCE
:GooglePlayGamesManifest.plugin:packageReleaseRenderscript NO-SOURCE
:compileReleaseRenderscript UP-TO-DATE
:checkReleaseManifest UP-TO-DATE
:generateReleaseBuildConfig UP-TO-DATE
:prepareLintJar UP-TO-DATE
:generateReleaseResValues UP-TO-DATE
:generateReleaseResources UP-TO-DATE
:GoogleMobileAdsPlugin:compileReleaseRenderscript
:GoogleMobileAdsPlugin:generateReleaseResValues
:GoogleMobileAdsPlugin:generateReleaseResources
:GoogleMobileAdsPlugin:packageReleaseResources
:GooglePlayGamesManifest.plugin:compileReleaseRenderscript
:GooglePlayGamesManifest.plugin:generateReleaseResValues
:GooglePlayGamesManifest.plugin:generateReleaseResources
:GooglePlayGamesManifest.plugin:packageReleaseResources
:mergeReleaseResources UP-TO-DATE
:createReleaseCompatibleScreenManifests UP-TO-DATE
:processReleaseManifest
:splitsDiscoveryTaskRelease UP-TO-DATE
:GoogleMobileAdsPlugin:platformAttrExtractor
:GooglePlayGamesManifest.plugin:platformAttrExtractor
:GoogleMobileAdsPlugin:generateReleaseBuildConfig
:GoogleMobileAdsPlugin:prepareLintJar UP-TO-DATE
:GoogleMobileAdsPlugin:javaPreCompileRelease
:GoogleMobileAdsPlugin:processReleaseJavaRes NO-SOURCE
:GooglePlayGamesManifest.plugin:generateReleaseBuildConfig
:GooglePlayGamesManifest.plugin:prepareLintJar UP-TO-DATE
:GooglePlayGamesManifest.plugin:javaPreCompileRelease
:GooglePlayGamesManifest.plugin:processReleaseJavaRes NO-SOURCE
:compileReleaseNdk NO-SOURCE
:mergeReleaseShaders UP-TO-DATE
:compileReleaseShaders UP-TO-DATE
:generateReleaseAssets UP-TO-DATE
:GoogleMobileAdsPlugin:mergeReleaseShaders
:GoogleMobileAdsPlugin:compileReleaseShaders
:GoogleMobileAdsPlugin:generateReleaseAssets
:GoogleMobileAdsPlugin:mergeReleaseAssets
:GooglePlayGamesManifest.plugin:mergeReleaseShaders
:GooglePlayGamesManifest.plugin:compileReleaseShaders
:GooglePlayGamesManifest.plugin:generateReleaseAssets
:GooglePlayGamesManifest.plugin:mergeReleaseAssets
:mergeReleaseAssets
:mergeReleaseJniLibFolders UP-TO-DATE
:GoogleMobileAdsPlugin:compileReleaseNdk NO-SOURCE
:GoogleMobileAdsPlugin:mergeReleaseJniLibFolders
:GoogleMobileAdsPlugin:transformNativeLibsWithMergeJniLibsForRelease
:GoogleMobileAdsPlugin:transformNativeLibsWithIntermediateJniLibsForRelease
:GooglePlayGamesManifest.plugin:compileReleaseNdk NO-SOURCE
:GooglePlayGamesManifest.plugin:mergeReleaseJniLibFolders
:GooglePlayGamesManifest.plugin:transformNativeLibsWithMergeJniLibsForRelease
:GooglePlayGamesManifest.plugin:transformNativeLibsWithIntermediateJniLibsForRelease
:processReleaseJavaRes NO-SOURCE
:validateSigningRelease
:GoogleMobileAdsPlugin:mergeReleaseConsumerProguardFiles UP-TO-DATE
:GoogleMobileAdsPlugin:transformNativeLibsWithSyncJniLibsForRelease
:GoogleMobileAdsPlugin:mergeReleaseResources
:GoogleMobileAdsPlugin:verifyReleaseResources
:GooglePlayGamesManifest.plugin:mergeReleaseConsumerProguardFiles UP-TO-DATE
:GooglePlayGamesManifest.plugin:transformNativeLibsWithSyncJniLibsForRelease
:GooglePlayGamesManifest.plugin:mergeReleaseResources
:GooglePlayGamesManifest.plugin:verifyReleaseResources
:GooglePlayGamesManifest.plugin:processReleaseResources
:GooglePlayGamesManifest.plugin:generateReleaseSources
:GooglePlayGamesManifest.plugin:compileReleaseJavaWithJavac
:GooglePlayGamesManifest.plugin:transformClassesAndResourcesWithPrepareIntermediateJarsForRelease
:GooglePlayGamesManifest.plugin:extractReleaseAnnotations
:GooglePlayGamesManifest.plugin:transformResourcesWithMergeJavaResForRelease
:GooglePlayGamesManifest.plugin:transformClassesAndResourcesWithSyncLibJarsForRelease
:GooglePlayGamesManifest.plugin:bundleRelease
:GoogleMobileAdsPlugin:processReleaseResources
:processReleaseResources
:generateReleaseSources
:GoogleMobileAdsPlugin:generateReleaseSources
:GoogleMobileAdsPlugin:compileReleaseJavaWithJavac
:GoogleMobileAdsPlugin:transformClassesAndResourcesWithPrepareIntermediateJarsForRelease
:javaPreCompileRelease
:compileReleaseJavaWithJavac
:compileReleaseSources
:lintVitalRelease
:transformClassesWithPreDexForRelease
:transformDexWithDexForRelease FAILED
72 actionable tasks: 56 executed, 16 up-to-date
]
exit code: 1
UnityEditor.Android.Command.WaitForProgramToRun (UnityEditor.Utils.Program p, UnityEditor.Android.WaitingForProcessToExit waitingForProcessToExit, System.String errorMsg)
UnityEditor.Android.Command.Run (System.Diagnostics.ProcessStartInfo psi, UnityEditor.Android.WaitingForProcessToExit waitingForProcessToExit, System.String errorMsg)
UnityEditor.Android.AndroidJavaTools.RunJava (System.String args, System.String workingdir, System.Action`1 progress, System.String error)
UnityEditor.Android.GradleWrapper.Run (UnityEditor.Android.AndroidJavaTools javaTools, System.String workingdir, System.String task, System.Action`1 progress)
Rethrow as GradleInvokationException: Gradle build failed
UnityEditor.Android.GradleWrapper.Run (UnityEditor.Android.AndroidJavaTools javaTools, System.String workingdir, System.String task, System.Action`1 progress)
UnityEditor.Android.PostProcessor.Tasks.BuildGradleProject.Execute (UnityEditor.Android.PostProcessor.PostProcessorContext context)
UnityEditor.Android.PostProcessor.PostProcessRunner.RunAllTasks (UnityEditor.Android.PostProcessor.PostProcessorContext context)
UnityEngine.GUIUtility:ProcessEvent(Int32, IntPtr)
```
 
![](https://i.gyazo.com/757ae4ffccffc894d4057810d4fff423.png)

Unity version: 2018.2.1f1
GoogleAds: 3.15.0 (https://github.com/googleads/googleads-mobile-unity/releases)
GooglePlayGamesPlugin: 0.9.50

How can I correct it. Any ideas?

**Comment by Linkupdated** [comment]
We search for a long time about thoses issue only to find the best way is to make sure every aar is at the same version.

But the question is how to do that with the resolver. You need to get older version of the plugin until the resolver got the right aar version. I hope this saves you the days we lost on this.

Good luck.

**Comment by exzizt** [comment]
I tried adding Google AdMobs to my Unity project last night and it broke everything. I've even removed it since then to try and restore it, but it's still broken.

Now I'm getting this issue:

> 08-01 13:37:09.213 13376 13376 E AndroidRuntime: FATAL EXCEPTION: main
08-01 13:37:09.213 13376 13376 E AndroidRuntime: Process: com.exzizt.Birdie, PID: 13376
08-01 13:37:09.213 13376 13376 E AndroidRuntime: java.lang.Error: FATAL EXCEPTION [main]
08-01 13:37:09.213 13376 13376 E AndroidRuntime: Unity version     : 2018.2.0f2
08-01 13:37:09.213 13376 13376 E AndroidRuntime: Device model      : LGE LG-D852
08-01 13:37:09.213 13376 13376 E AndroidRuntime: Device fingerprint: lge/g3_tls_ca/g3:6.0/MRA58K/162031018df96:user/release-keys
08-01 13:37:09.213 13376 13376 E AndroidRuntime:
08-01 13:37:09.213 13376 13376 E AndroidRuntime: Caused by: java.lang.NoClassDefFoundError: Failed resolution of: Lcom/google/android/gms/common/api/Api$zzf;
08-01 13:37:09.213 13376 13376 E AndroidRuntime:        at com.google.android.gms.auth.api.Auth.<clinit>(Unknown Source)
08-01 13:37:09.213 13376 13376 E AndroidRuntime:        at com.google.games.bridge.TokenFragment.buildClient(TokenFragment.java:382)
08-01 13:37:09.213 13376 13376 E AndroidRuntime:        at com.google.games.bridge.TokenFragment.processRequest(TokenFragment.java:279)
08-01 13:37:09.213 13376 13376 E AndroidRuntime:        at com.google.games.bridge.TokenFragment.onResume(TokenFragment.java:506)
08-01 13:37:09.213 13376 13376 E AndroidRuntime:        at android.app.Fragment.performResume(Fragment.java:2272)
08-01 13:37:09.213 13376 13376 E AndroidRuntime:        at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1008)
08-01 13:37:09.213 13376 13376 E AndroidRuntime:        at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1148)
08-01 13:37:09.213 13376 13376 E AndroidRuntime:        at android.app.BackStackRecord.run(BackStackRecord.java:793)
08-01 13:37:09.213 13376 13376 E AndroidRuntime:        at android.app.FragmentManagerImpl.execPendingActions(FragmentManager.java:1535)
08-01 13:37:09.213 13376 13376 E AndroidRuntime:        at android.app.FragmentManagerImpl$1.run(FragmentManager.java:482)
08-01 13:37:09.213 13376 13376 E AndroidRuntime:        at android.os.Handler.handleCallback(Handler.java:739)
08-01 13:37:09.213 13376 13376 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:95)
08-01 13:37:09.213 13376 13376 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:148)
08-01 13:37:09.213 13376 13376 E AndroidRuntime:        at android.app.ActivityThread.main(ActivityThread.java:5551)
08-01 13:37:09.213 13376 13376 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
08-01 13:37:09.213 13376 13376 E AndroidRuntime:        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:730)
08-01 13:37:09.213 13376 13376 E AndroidRuntime:        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:620)
08-01 13:37:09.213 13376 13376 E AndroidRuntime: Caused by: java.lang.ClassNotFoundException: Didn't find class "com.google.android.gms.common.api.Api$zzf" on path: DexPathList[[zip file "/data/app/com.exzizt.Birdie-1/base.apk"],nativeLibraryDirectories=[/data/app/com.exzizt.Birdie-1/lib/arm, /data/app/com.exzizt.Birdie-1/base.apk!/lib/armeabi-v7a, /vendor/lib, /system/lib]]
08-01 13:37:09.213 13376 13376 E AndroidRuntime:        at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)
08-01 13:37:09.213 13376 13376 E AndroidRuntime:        at java.lang.ClassLoader.loadClass(ClassLoader.java:511)
08-01 13:37:09.213 13376 13376 E AndroidRuntime:        at java.lang.ClassLoader.loadClass(ClassLoader.java:469)
08-01 13:37:09.213 13376 13376 E AndroidRuntime:        ... 17 more
08-01 13:37:09.213 13376 13376 E AndroidRuntime:        Suppressed: java.lang.ClassNotFoundException: com.google.android.gms.common.api.Api$zzf
08-01 13:37:09.213 13376 13376 E AndroidRuntime:                at java.lang.Class.classForName(Native Method)
08-01 13:37:09.213 13376 13376 E AndroidRuntime:                at java.lang.BootClassLoader.findClass(ClassLoader.java:781)
08-01 13:37:09.213 13376 13376 E AndroidRuntime:                at java.lang.BootClassLoader.loadClass(ClassLoader.java:841)
08-01 13:37:09.213 13376 13376 E AndroidRuntime:                at java.lang.ClassLoader.loadClass(ClassLoader.java:504)
08-01 13:37:09.213 13376 13376 E AndroidRuntime:                ... 18 more
08-01 13:37:09.213 13376 13376 E AndroidRuntime:        Caused by: java.lang.NoClassDefFoundError: Class not found using the boot class loader; no stack trace available

**Comment by sptkjs** [comment]
Is solved it? I have same issue.

**Comment by ocveran** [comment]
While I solved this way: https://github.com/playgameservices/play-games-plugin-for-unity/issues/2203

**Comment by olehkuznetsov** [comment]
Should be fixed in 0.9.51.
